src/components/contacts/phone-number.ts

```diff
@@ -0,0 +1,159 @@
+import {t} from '@lingui/macro'
+import {
+  isSupportedCountry,
+  ParseError,
+  parsePhoneNumber,
+  parsePhoneNumberWithError,
+  type PhoneNumber,
+} from 'libphonenumber-js/max'
+
+import {type CountryCode} from '#/lib/international-telephone-codes'
+
+/**
+ * Intended for after the user has finished inputting their phone number.
+ */
+export function processPhoneNumber(
+  number: string,
+  country: CountryCode,
+):
+  | {
+      valid: true
+      formatted: string
+    }
+  | {
+      valid: false
+      reason?: string
+    } {
+  try {
+    const phoneNumber = parsePhoneNumberWithError(number, {
+      defaultCountry: country,
+    })
+    if (!phoneNumber.isValid()) {
+      return {valid: false, reason: t`Invalid phone number`}
+    }
+    const type = phoneNumber.getType()
+    if (
+      type !== 'MOBILE' &&
+      type !== 'FIXED_LINE_OR_MOBILE' &&
+      type !== 'PERSONAL_NUMBER'
+    ) {
+      return {
+        valid: false,
+        reason: t`Number should be a mobile number`,
```

Comment:
```suggestion
        reason: t`Number should be a cellular number`,
```
micronit: When I was living in the US (admittedly rather a long time ago), cellular/cell/cellphone was more commonly used. No idea if that's still the case though 🤷🏻‍♂️